src/parseMetaboliteInfoFromSBML.py

```diff
@@ -5,113 +5,101 @@
+        modelDoc:libsbml.SBMLDoc - a Path object to the sbml file.
+        strict:bool - Whether to raise an error or just a warning. Default: False - only raise Warning.
     '''
-    # handle non-existing files as sbml won't complain
-    if not modelfile.exists():
-        raise FileExistsError(str(modelfile) + " does not exist")
-    # read the sbml file
-    reader = sbml.SBMLReader()
-    doc = reader.readSBMLFromFile(modelfile)
-    #doc = reader.readSBMLFromFile("./Recon3DModel_301.xml")
-    mod = doc.getModel()
-    # go through the metabolites in the model and extract the relevant information
-    metabolites = mod.getListOfSpecies() 
-
-    met_dic = {"ID":[],
-            "Name":[],
-            "Charge":[],
-            "Compartment":[],
-            "Chemical_formula":[],
-            "Annotation_source":[],
-            "Annotation_value":[]}
-    for met_i in range(len(metabolites)):
-        # get annotations first, if there are more than 1, use the length as multiplier
-        met = mod.getSpecies(met_i)
-        urls = met.getAnnotationString().split("\n")
-        if len(urls) > 0:
-            anno_source = [x.split("/")[3].strip('"') for x in urls if re.search("identifiers.org",x) != None]
-            met_dic["Annotation_source"].extend(anno_source)
-            met_dic["Annotation_value"].extend( ["/".join(x.split("/")[4:]).replace('"/>',"") for x in urls if re.search("identifiers.org",x) != None])
-            mult = len(anno_source)
+    validator = sbml.SBMLValidator()
+    errors = validator.validate(modelDoc)
+    if errors > 0:
+        text = '''{MSG}: SBML file not valid, following failures have been found:\n''' + validator.getErrorLog().printErrors()
+        if strict:
+            raise IOError(text.format(MSG = "ERROR"))
         else:
-            mult = 1
-            met_dic["Annotation_source"].extend([""] * mult)
-            met_dic["Annotation_value"].extend( [""]* mult)
-        
-        # get the rest of the information
-        met_dic["ID"].extend([met.getId()] * mult)
-        met_dic["Name"].extend([met.getName()] * mult)
-        met_dic["Charge"].extend([met.getCharge()] * mult)
-        met_dic["Compartment"].extend([met.getCompartment()] * mult)
-        
-        # finally look for the chemical formula
-        fbc = met.getPlugin("fbc")
-        met_dic["Chemical_formula"].extend([fbc.getChemicalFormula()] * mult)
-
-        # debugging
-       # nrows = [len(x) for x in met_dic.values()]
-       # if not all(x == nrows[0] for x in nrows):
-       #     raise 
-       #     break
+            warning.warn(text.format(MSG = "WARNING"))
     
-    # cast the dictionary into a data frame and return it
-    df = pd.DataFrame(met_dic)
-    return(df)
+def annotateMetabolites_bulk(metabolites:list[MeMoMetabolite]) -> None:
+    '''
+    This function tries to get the InChI string for each metabolite by leveraging the information in the annotation attribute of the metabolite
+    '''
+
 
```

Comment:
I don't understand where the definition for the annotateMetabolites_bulk() function ends @Porthmeus

Reply:
It is still a stub - I already added some functionality, in the branch locally, but I did not want to include it in the current pull request. The next iteration will include a MeMoModel object which implements the bulk annotation of metabolites - while there will still be the annotate function for individual metabolites. This distinction is necessary because we want to reduce the number of requests to different databases. Thus, everything we can annotate in a bulk manner will be implemented in the annotateMetabolite_bulk() function as part of the MeMoModel object. Everything which requires individual requests to databases will be implemented in the annoatate() function of the MeMoMetabolite object.